packages/react/react.ts

```diff
@@ -48,13 +50,15 @@ const isValidElement = (vnode?: VNode | VNode[] | null) => {
   return false;
 };
 
-const memo = (
-  component: (...args: unknown[]) => VNode
-) => {
+const memo = (component: (...args: unknown[]) => VNode) => {
+  const cache = new Map();
   return (props: VProps) => {
-    return component(Object.values(props));
+    if (!cache.has(props)) return cache.get(props);
```

Comment:
```suggestion
    if (cache.has(props)) return cache.get(props);
```

Also IMHO the `React.memo` is using shallow compare by default (while `Map.prototype.has` is using `SameValueZero` which compares the reference equality).

Reply:
This implementation of memoization will create a memory leak because there is no mechanism for clearing the cache. React also memoizes only the last render.

Reply:
I see, thank you for the feedback. Let me adjust the code to fix